### PR TITLE
feat: simplify maintenance

### PR DESCRIPTION
**Pull request by jobulcke**

### CODE REVIEW

File: docs/how-to-run.md [hunk 1]
```diff
@@ -126,28 +126,16 @@ Here is an explanation provided for all the possibilities on how to tweak and co
...
-    <td>Defines how often the redundant compacted fragments will be checked for deletion<sup>3</sup></td>
-    <td>No</td>
-    <td>0 0 0 * * *</td>
-  </tr>
-  <tr><td colspan="4"><b>Retention (<a href="./configuration/retention-policies">Retention Policies</a>)</b></td></tr>
-  <tr>
-    <td>ldes-server.retention-cron</td>
-    <td>Defines how often the Retention Service will check the members<sup>3</sup></td>
+    <td>ldes-server.maintenance-cron</td>
+    <td>Defines how often the maintenance job will run, which includes <a href="./configuration/retention-policies">retention</a>, compaction and deletion<sup>3</sup></td>
```
Comment (Yalz) [inline docs/how-to-run.md:138]: could also then link to Compaction..? https://informatievlaanderen.github.io/VSDS-LDESServer4J/3.4.0-SNAPSHOT/features/compaction

File: ldes-server-maintenance/ldes-server-compaction/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/compaction/batch/CompactionCandidateSorter.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+					splitList.add(currentCandidate.get());
+				}
+				long nextId = currentCandidate.get().getNextPageId();
+				currentCandidate = candidatesList.stream().filter(c -> c.getId() == nextId).findFirst();
+			}
+
+			if (splitList.size() > 1) {
+				compactionDesign.put(index.incrementAndGet(), splitList);
+			}
+		});
```
Comment (Yalz) [inline ldes-server-maintenance/ldes-server-compaction/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/compaction/batch/CompactionCandidateSorter.java:42]: I find this method quite builky.. Maybe split some things off into smaller private methods if possible ?

File: ldes-server-maintenance/ldes-server-compaction/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/compaction/batch/CompactionDbWriter.java [hunk 3]
```diff
@@ -12,27 +13,30 @@
 import java.util.Set;
 
 @Component
-public class PaginationCompactionService {
+public class CompactionDbWriter {
```
Comment (Yalz) [inline ldes-server-maintenance/ldes-server-compaction/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/compaction/batch/CompactionDbWriter.java:16]: Personally (so this does not mean you SHOULD), I would leave out the "DB". Writing implies already a datasource interaction

File: ldes-server-maintenance/ldes-server-compaction/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/compaction/batch/CompactionStepDefinitions.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.batch.core.step.builder.StepBuilder;
+import org.springframework.batch.core.step.tasklet.Tasklet;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.task.SimpleAsyncTaskExecutor;
+import org.springframework.transaction.PlatformTransactionManager;
+
+@Configuration
+public class CompactionStepDefinitions {
+	public static final String COMPACTION_STEP = "partitionedCompactionStep";
```
Comment (Yalz) [inline ldes-server-maintenance/ldes-server-compaction/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/compaction/batch/CompactionStepDefinitions.java:15]: Would personally drop the partitioned. Does not provide extra info when reading jobs/steps

File: ldes-server-maintenance/ldes-server-maintenance-common/pom.xml [hunk 5]
```diff
@@ -11,6 +11,40 @@
...
+            <artifactId>junit-jupiter</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+        </dependency>
```
Comment (Yalz) [inline ldes-server-maintenance/ldes-server-maintenance-common/pom.xml:35]: test scope please

File: ldes-server-maintenance/ldes-server-maintenance-common/src/main/java/module-info.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+open module ldes.server.maintenance.common {
+	requires spring.context;
+	requires spring.batch.core;
+	requires ldes.domain;
+	requires spring.beans;
+	exports be.vlaanderen.informatievlaanderen.ldes.server.maintenance.services;
+	exports be.vlaanderen.informatievlaanderen.ldes.server.maintenance.valueobjects;
+}
```
Comment (Yalz) [inline ldes-server-maintenance/ldes-server-maintenance-common/src/main/java/module-info.java:8]: Bold to keep the module infos alive 😄

**Review by Yalz — APPROVED** [review]

File: ldes-server-maintenance/ldes-server-maintenance-common/pom.xml (hunk 5 above)
Comment (jobulcke) [reply ldes-server-maintenance/ldes-server-maintenance-common/pom.xml:35]: Is test scoped by the dependencyManagement of the parent 🙂

**Review by Yalz — APPROVED** [review]

**Pull request merged**